src/train_cmd.cpp

```diff
@@ -1662,19 +1662,31 @@ static bool TrainApproachingCrossing(TileIndex tile)
 	return HasVehicleOnPos(tile_from, &tile, &TrainApproachingCrossingEnum);
 }
 
+/** Check if the crossing should be closed
+ *  @return train on crossing || train approaching crossing || reserved
+ */
+static inline bool CheckLevelCrossing(TileIndex tile)
```

Comment:
Name is way to ambiguous. It is checking for what? Is it checking if it is a level crossing? If it should bar? So many questions :D